packages/core/src/modules/oob/messages/OutOfBandInvitation.ts

```diff
@@ -103,7 +103,7 @@ export class OutOfBandInvitation extends AgentMessage {
   // TODO: this only takes into account inline didcomm services, won't work for public dids
   public getRecipientKeys(): Key[] {
     return this.services
-      .filter((s): s is OutOfBandDidCommService => typeof s !== 'string')
+      .filter((s): s is OutOfBandDidCommService => typeof s !== 'string' && !(s instanceof String))
       .map((s) => s.recipientKeys)
```

Comment:
Just a comment. The error mentioned in https://github.com/hyperledger/aries-framework-javascript/issues/812 actually came from the `...curr`, because the previous map returns `[undefined]` in case of public did is used in the invitation.

So I'd suggest to modify the second mapping something like this:
```suggestion
      .map((s) => s.recipientKeys || [])
```

Reply:
Hmm the filter method should filter out all non-string values, so it shouldn't be possible there is a type undefined in the array. It think by adding this to the filter it won't have the undefined anymore.

Reply:
`s.recipientKeys` was undefined. The json parser converts the string did into some weird object, maybe that will get resolved in the future though.